### ISSUE TITLE
Feature request: sst remove --force

### ISSUE THREAD

**Issue by agamm** [anomalyco/sst#5199]
If we accidently stop or if something crashes while doing an `sst remove` it might block removing the resource when trying to remove again, even an `sst unlock` won't help.

I got errors like:
> DeleteOriginAccessControl, https response error StatusCode: 409

I had to login to AWS and remove the resources manually (I hate the AWS UI), and then `sst refresh && sst remove`.

It would be great to have a `--force` command to do all that via sst, could that be added?

**Comment by JessicaSachs** [comment]
:+1: Having to go to the AWS UI right now (and sad about it)

**Comment by jayair** [comment]
So would the force flag ignore the missing resources and continue on with removing the rest?

**Comment by agamm** [comment]
I would like it to remove all the resources it can find of that stage (maybe via tags?). And force remove them so I won't have to manually login to the UI to search them one by one and figure out what is dependant on what.

**Comment by JessicaSachs** [comment]
Yeah that's what I'd expect. I'd expect to log out the already deleted resources and then continue removing everything else. If there's disparity in the local sst-env.d.ts it would get fully cleaned up.

**Comment by jayair** [comment]
Yeah I can see that. cc @fwang

**Comment by fwang** [comment]
Sharing some context from the conversation w/ @agamm on Discord:

> DeleteOriginAccessControl, https response error StatusCode: 409

This error happens when the distribution is in the `Deploying` state, meaning CloudFront is in the middle of making changes to the distribution. And the distribution cannot be removed while in this state.. SST can't remove it and u can't remove it from AWS Console.

We've seen distributions getting stuck in this state. And it can take CloudFront quite long to finish whatever it is doing.

By the time @agamm was able to remove it in AWS Console, the distribution should've become in a stable state. Running `sst remove` at the time would've also be able to remove it.

If u can delete something in the AWS Console, `sst remove` will be also be able to delete it. The idea of a "force deploy" doesn't really apply.

Hope that makes sense.

**Comment by agamm** [comment]
> Sharing some context from the conversation w/ @agamm on Discord:
> 
> > DeleteOriginAccessControl, https response error StatusCode: 409
> 
> This error happens when the distribution is in the `Deploying` state, meaning CloudFront is in the middle of making changes to the distribution. And the distribution cannot be removed while in this state.. SST can't remove it and u can't remove it from AWS Console.
> 
> We've seen distributions getting stuck in this state. And it can take CloudFront quite long to finish whatever it is doing.
> 
> By the time @agamm was able to remove it in AWS Console, the distribution should've become in a stable state. Running `sst remove` at the time would've also be able to remove it.
> 
> If u can delete something in the AWS Console, `sst remove` will be also be able to delete it. The idea of a "force deploy" doesn't really apply.
> 
> Hope that makes sense.

So the `--force` remove still makes sense, right? I would try to use the tagmanager api to find the relevant tags up and running.

PS more reasons for `--force` https://discord.com/channels/983865673656705025/1317480173763039262 and https://discord.com/channels/983865673656705025/1317268467916738621/1317288380937011240

UPDATE: it happended again (remove -> dev (stuck -> ctrl+c) -> got it again - even after 10 minute wait):
```
pnpx sst refresh --stage agam
SST 3.4.5  ready!

➜  App:        myapp
   Stage:      agam

~  Refresh

|  Locked      A concurrent update was detected on the app. Run `sst unlock` to remove the lock and try again.
```

**Comment by fwang** [comment]
> I would try to use the tagmanager api to find the relevant tags up and running.

Interesting! Just tried running this and it returned a bunch of resources.
```bash
aws resourcegroupstaggingapi get-resources --tag-filters Key=sst:app,Values=playground
```

Is this the tag manager api u were talking about? Does it support all aws resource types?

**Comment by agamm** [comment]
> > I would try to use the tagmanager api to find the relevant tags up and running.
> 
> Interesting! Just tried running this and it returned a bunch of resources.
> 
> ```shell
> aws resourcegroupstaggingapi get-resources --tag-filters Key=sst:app,Values=playground
> ```
> 
> Is this the tag manager api u were talking about? Does it support all aws resource types?

Yeah, I think it's also the best way we have in AWS to delete resources we don't know about. 

EDIT: I think it would be also nice to see what the tag manager found before you confirm removal.

**Comment by gtokman** [comment]
@fwang any updates on adding this flag? It makes it difficult to remove resources if you get errors during that process.

We are seeing errors such as:
```
✕  Failed    
   web-docs sst:aws:Nextjs → web-docsAssetFiles sst:aws:BucketFiles
   operation error S3: PutObject, https response error StatusCode: 404, RequestID: TSR2Y5HZ8PPVBFJV, HostID: cnNbkREBg0P+4mJ8p5Vj2dMQOCFm8iZdHWWfVfAh5HPmklzjnbO/r9zNj2unE+MB1kpxT4d8z+c+gf5t8SazO+6lie696Mh6, api error NoSuchBucket: The specified bucket does not exist
✕  Failed    
   web-docs sst:aws:Nextjs → web-docsAssetFiles sst:aws:BucketFiles
   operation error S3: DeleteObject, https response error StatusCode: 404, RequestID: PXRY9TDFC5ZZ1JSE, HostID: sH9RYa3Xw2Bne/nbXyPd0gvLIWAherYzoTozCWS2mwhaTiWnE7uIqZHsOYKPB6F97/lx1JcMeXg=, api error NoSuchBucket: The specified bucket does not exist

   web-docs sst:aws:Nextjs → web-docsS3AccessControl sst:aws:OriginAccessControl
   operation error CloudFront: GetOriginAccessControl, https response error StatusCode: 404, RequestID: b74756f8-7644-4247-9409-f772c09574bc, NoSuchOriginAccessControl: The specified origin access control does not exist.

   web-portal sst:aws:Nextjs → web-portalS3AccessControl sst:aws:OriginAccessControl
   operation error CloudFront: DeleteOriginAccessControl, https response error StatusCode: 409, RequestID: 8036ceba-31fb-45f8-aad4-690b630fc523, OriginAccessControlInUse: The CloudFront origin access control is still being used.

   web sst:aws:Nextjs → webS3AccessControl sst:aws:OriginAccessControl
   operation error CloudFront: DeleteOriginAccessControl, https response error StatusCode: 409, RequestID: 38b4613a-6e12-4658-bee6-77f2a4632831, OriginAccessControlInUse: The CloudFront origin access control is still being used.
~``

**Comment by jayair** [comment]
@gtokman Can you tell me the steps you ran that caused those error?

**Comment by gtokman** [comment]
@jayair I'm not 100% sure tbh. 

We use a shared stage for open PRs, but when multiple PRs are opened simultaneously, CI sometimes attempts to deploy to the same stage concurrently. This usually results in an error prompting us to run sst unlock. We usually re-kick the job, and it typically succeeds.

However, there are times when this error occurs, and things don’t resolve as expected, leading to further issues.

**Comment by jayair** [comment]
As an aside, it would be better to not have multiple deploys run on the same stage in your CI.

**Comment by agamm** [comment]
I got a problem where this will really help trying to `sst remove` and got:
```
Redis sst:aws:Redis → RedisSubnetGroup aws:elasticache:SubnetGroup
sdk-v2/provider2.go:515: sdk.helper_schema: deleting ElastiCache Subnet Group (apprais-v2-agam-redissubnetgroup-mwcnktdu): operation error ElastiCache: DeleteCacheSubnetGroup, https response error StatusCode: 400, RequestID: 542d3c14-8dec-4498-b672-f4d375a98a63, CacheSubnetGroupInUse: Cache subnet group myproject-agam-redissubnetgroup-mwcnktdu is currently in use by a cache cluster.: provider=aws@6.66.2
```

**Comment by ksmsk** [comment]
+1 for this feature. i literally had spent my whole afternoon trying to remove a pesky hanging stage. this is what i had done:
1. if there are any authentication errors about some resources while removing it, go to aws console and delete those resources manually.
2. run `sst refresh --stage {stage}`
3. run `sst deploy --stage {stage}`
(if it's created by your CI/CD and you have variables in the pipeline, just mock those variables and tweak your sst config to make it deployable from your local env.)
4. now you can `sst remove --stage {stage}`

hopefully you can get rid off those pesky stages

**Comment by dx-nhr157** [comment]
there is no way for me to progress once I receive a `ConcurrentOperationException` . I am unable to delete the stack anymore. ANd i receive this error because it gets stuck deleting a custom cognito auth domain.

**Comment by fwang** [comment]
Just read up on this thread, so 2 types of errors were brought up in this thread:

1. Resource already removed in AWS, and SST tries to remove it again and fails. For example:
    > operation error S3: PutObject, https response error StatusCode: 404, RequestID: TSR2Y5HZ8PPVBFJV, HostID: cnNbkREBg0P+4mJ8p5Vj2dMQOCFm8iZdHWWfVfAh5HPmklzjnbO/r9zNj2unE+MB1kpxT4d8z+c+gf5t8SazO+6lie696Mh6, api error NoSuchBucket: The specified bucket does not exist

    Run `sst refresh` will fix this. When u run `sst refresh`, SST will go through each resource in ur SST app state and cross check the current state of that resource in AWS. If it's not found, the resource will be removed from the state. After refresh, run `sst remove` again.

2. Resource still in use
    > operation error CloudFront: DeleteOriginAccessControl, https response error StatusCode: 409, RequestID: 8036ceba-31fb-45f8-aad4-690b630fc523, OriginAccessControlInUse: The CloudFront origin access control is still being used.

    The resource is still in use. It does not matter how u try to remove it (AWS console, AWS CLI, etc), remove will fail with the same error. Why this happens can vary based on the type of resource. But adding a `--force` won't help.

**Comment by agamm** [comment]
> * Resource already removed in AWS, and SST tries to remove it again and fails. For example:
>   > operation error S3: PutObject, https response error StatusCode: 404, RequestID: TSR2Y5HZ8PPVBFJV, HostID: cnNbkREBg0P+4mJ8p5Vj2dMQOCFm8iZdHWWfVfAh5HPmklzjnbO/r9zNj2unE+MB1kpxT4d8z+c+gf5t8SazO+6lie696Mh6, api error NoSuchBucket: The specified bucket does not exist
>   
>   
>   Run `sst refresh` will fix this. When u run `sst refresh`, SST will go through each resource in ur SST app state and cross check the current state of that resource in AWS. If it's not found, the resource will be removed from the state. After refresh, run `sst remove` again.
> 
> *

1. You can automatically refresh when using `--force` this will help newer users that are stuck. And is easy to implement.
2. Removing dependencies by looking for tags helps, this is a cumbersome manual process that would be much better handled automatically.
